static/app/views/dashboardsV2/gridLayout/dashboard.tsx

```diff
@@ -221,33 +234,51 @@ class Dashboard extends Component<Props> {
           isEditing={isEditing}
           onDelete={this.handleDeleteWidget(widget)}
           onEdit={this.handleEditWidget(widget, index)}
+          hideDragHandle={isMobile}
         />
       </GridItem>
     );
   }
 
+  onLayoutChange = newLayout => {
+    const {onLayoutChange} = this.props;
+
+    // Check component width because onLayoutChange can trigger
+    // before setting mobile breakpoint state
+    const rect = document.querySelector('.react-grid-layout')?.getBoundingClientRect();
```

Comment:
I don't really like this. I was trying to use `React.createRef` earlier and assign it to a div wrapping `GridLayout` so I could check `this.gridContainer.current.offsetWidth` but it was a huge pain to test it. I'm wondering what we think of this because it's more simple than a ref, but less elegant looking.

Reply:
Hmm Grid Layout has a handler called `onBreakpointChange`, maybe you could use that to manage the `isMobile` state instead? Might or might not help with this particular problem

Reply:
Found another way to set the layout at the right time!